app/helpers/application_helper.rb

```diff
@@ -7,4 +7,8 @@ def crumb_li(title, path, active)
       active ? title : link_to(title, path)
     end
   end
+
+  def overwrites_existing?(short_url_request)
```

Comment:
I don't feel too strongly but I wonder if `would_overwrite_existing?` might be a small readability improvement for the method name here. My thinking is that at the point when this is called, the request isn't confirmed so it doesn't overwrite an existing request.